src/main/resources/xml/schema/2020-1/nordic2020-1.nav-references.sch

```diff
@@ -125,9 +125,9 @@
             <let name="document-in-nav" value="((preceding::html:a | self::*) intersect ancestor::html:nav//html:a)[substring-before(@href,'#') = $href][1]"/>
             <let name="document-in-nav-depth" value="count($document-in-nav/ancestor::html:li)"/>
             <let name="depth-in-nav" value="count(ancestor::html:li)"/>
-            <let name="depth-in-content" value="$result-ref-first/xs:integer((@data-heading-depth, @data-sectioning-depth)[1])"/>
```

Comment:
First a thing I notice now about `@data-sectioning-depth`, that might cause problems:

If the part is contained in a separate content file (which I think is the most common case), then we should add one to `data-sectioning-depth`. So: find the first first preceding `c:result` with `@data-heading-element='h1'`. If it is a part according to either `@data-document-epub-type` or `@data-document-role`, then add one to `data-sectioning-depth`. This assumes that the part has a `h1` element, which might not be a good assumption? But if we assume that it does have a `h1`, we could do something like:

```xml
<let name="data-sectioning-depth"
     value="$result-ref-first/xs:integer(@data-sectioning-depth)
            + if ((preceding::c:result[@data-heading-element='h1'])[1][
                  @data-document-epub-type = 'part'
                  or @data-document-role = 'doc-part'
            ]) then 1 else 0"/>
```

A better solution might be to do a `p:wrap-sequence` before `<p:for-each name="epub3-validate.step.group-nav-references.iterate-html">` in `…/xproc/step/2020-1/epub3-validate.step.xpl`. Then the whole book would be in the context before running `list-heading-and-pagebreak-references.xsl` so that the `@data-sectioning-depth` could be correctly calculated there. If performance becomes a problem when wrapping all the content, we could `p:delete` all irrelevant elements before wrapping and running the XSLT.

Reply:
Hi @josteinaj 

Thank you for the example. It needed some minor changes and a lot of testing but the general concept was helpful.

I don't know if the `data-document-role` will be set and if so could be multiple values. If not this should work just fine.

Best regards
Daniel